pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -30,6 +41,12 @@ func (lp *LifecycleProcessor) OnInvokeStart(startDetails *InvocationStartDetails
 	log.Debug("[lifecycle] ---------------------------------------")
 
 	if !lp.DetectLambdaLibrary() {
+		if InferredSpansEnabled {
+			log.Debug("[lifecycle] Attempting to create inferred span")
+			inferredSpan = inferredspan.GenerateInferredSpan(startDetails.StartTime)
+			inferredspan.RouteInferredSpan(startDetails.InvokeEventRawPayload, inferredSpan)
```

Comment:
Let's refactor this into two steps:
- A call to `ParseEventSource`, which takes the event payload and the lambda context, and returns the event source (this will be useful in the future when adding other functionality based on event source, e.g. trigger types)
- A call to `RouteInferredSpan` which takes and event source and routes the inferred span to appropriate enricher